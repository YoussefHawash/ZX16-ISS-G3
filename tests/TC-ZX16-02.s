addi x1 ,45  #x1=45    0000 0000 0010 1101
li x2,33     #x2 	0000 0000 0010 0001
li x3 ,62
sltui x2 ,30  #x2 = 0000 0000 0000 0001
neg x2        #x2 = 1111 1111 1111 1101  -3
slli x2 ,3    #x2=  1111 1111 1110 1000  -8
srai x3,  4    #x3 = 0000 0000 0000 0100  4
ori  x1 , 22   #x1=  	0000 0000 0001 1111 31
andi x3, 50  	#x3 =0000 0000 0000 0100 4
xori x2, 55	#x2 = 1111 1111 1100 1000 65480
clr x2          #x2= 0000 0000  0000 0000
ecall 0x008
# x0 =0 x1= 31 x2= 0 x3=4 
ecall 0x00A
