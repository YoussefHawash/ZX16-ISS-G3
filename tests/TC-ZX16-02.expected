ADDi x1 ,45  
LI x2,33    
LI x3 ,77
SLTUI x2 ,30  
XORI x2, -1       
ADDI x2, 1       
SLLI x2 ,3    
SRAI x3  4    
ORI  x1 , 22   
ANDI x3, 85  
XORI x2, 55	
XOR x2, x2       
ECALL 0x008
ECALL 0x00A